eth/protocols/snap/handler.go

```diff
@@ -469,7 +469,7 @@ func ServiceGetByteCodesQuery(chain *core.BlockChain, req *GetByteCodesPacket) [
 			// Peers should not request the empty code, but if they do, at
 			// least sent them back a correct response without db lookups
 			codes = append(codes, []byte{})
-		} else if blob, err := chain.ContractCodeWithPrefix(hash); err == nil {
+		} else if blob, err := chain.ContractCode(hash); err == nil {
```

Comment:
I understand that this is a fallback.

`ContractCode()` will eventually call `ReadCode()`. `ReadCode()` will first try to read code with prefix; using key as `codeKey(hash)`. If failed, fallback by reading code using key as `hash`.

Can we add this details in PR description?